java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -602,6 +605,199 @@ public static CompletableFuture<Object> arrinsert(
+     *     or arrays.
+     * @param start The starting index, otherwise it will be defaulted to `0`.
+     * @return An array of integers. Each value is the index of the matching element in the array at
+     *     the path. The value is -1 if not found. If there are no values in the array, the return
+     *     value will be <code>null</code>
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "{\"a\": [\"value\", 3], \"b\": {\"a\": [3, [\"value\", false], 5]}}").get();
+     * Long[] result = Json.arrindex(client, gs(key), gs("$..a"), new JsonScalar(3), 1).get();
+     * assertArrayEquals(new Long[] {Long.valueOf(1), Long.valueOf(-1)}, result);
+     * }</pre>
+     */
+    public static CompletableFuture<Long[]> arrindex(
+            @NonNull BaseClient client,
+            @NonNull GlideString key,
+            @NonNull GlideString path,
+            @NonNull JsonScalar scalar,
+            @NonNull Integer start) {
+        return Json.<Object[]>executeCommand(
+                        client,
+                        new GlideString[] {
+                            gs(JSON_ARRINDEX), key, path, gs(scalar.toString()), gs(start.toString())
+                        })
+                .thenApply(ret -> castArray(ret, Long.class));
+    }
+
+    /**
+     * Searches for the first occurrence of a <code>scalar</code> JSON value in the arrays at the
+     * path.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @param path The path within the JSON document.
+     * @param scalar The scalar value to search for; JSON scalar refers to values that are not objects
+     *     or arrays.
+     * @param start The starting index, otherwise it will be defaulted to `0`.
+     * @param end The ending index. It will default to `0` if not provided. The last element is
+     *     included. `0` or `-1` means the last element is included.
+     * @return An array of integers. Each value is the index of the matching element in the array at
+     *     the path. The value is -1 if not found. If there are no values in the array, the return
+     *     value will be <code>null</code>
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "{\"a\": [\"value\", 3, 3], \"b\": {\"a\": [3, [\"value\", false], 5]}}").get();
+     * Long[] result = Json.arrindex(client, key, "$..a", new JsonScalar(3), 2, 4).get();
+     * assertArrayEquals(new Long[] {Long.valueOf(2), Long.valueOf(-1)}, result);
+     * }</pre>
+     */
+    public static CompletableFuture<Long[]> arrindex(
+            @NonNull BaseClient client,
+            @NonNull String key,
+            @NonNull String path,
+            @NonNull JsonScalar scalar,
+            @NonNull Integer start,
+            @NonNull Integer end) {
+        return arrindex(client, gs(key), gs(path), scalar, start, end);
+    }
+
```

Comment:
Consider creating arrindex options class, so you can reduce number of signature being added.
Keep in mind, start and end are both `long` (lowercase)